### ISSUE TITLE
GameGuru MAX - Bug - UV Material Issue

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#2353]
UV maps are not displaying correctly on all models. As a result, materials are stretching a bit at the edges and this is causing an issue for some of my models where I am making tiles that need to be seamless. I will try to illustrate. Here's a floor tile in the MAX importer:

![tile1](https://user-images.githubusercontent.com/36735507/156332873-ea499c21-46a6-42f7-856b-70711ba3b074.png)

If you look at the edges where the black diamond pattern meets the metallic area on the right, it's perfect. The diamond pattern continues perfectly until it meets the shiny metal, just as it's supposed to, just as I designed it to. So, the importer is displaying my model correctly. Now, let's look at this model once it actually gets into MAX:

![tile2](https://user-images.githubusercontent.com/36735507/156333174-c58d76c9-cd48-4e48-b3cb-8b3e7cf6d271.png)

You may need to expand the image to full size to see what's going on, but look within the areas I've circled in red. The diamond pattern is stretched as if the pattern has shifted or been squished a bit. And the metal texture is streaked as if it ended a few pixels away from the end and was stretched, too. Here's another angle:

![tile3](https://user-images.githubusercontent.com/36735507/156333514-c23d8d79-aa7b-4ef2-837c-b2e64798b3ca.png)

The result is, when I put two of these together, they don't tile like they were designed to tile:

![tile4](https://user-images.githubusercontent.com/36735507/156333633-e280c745-b8fd-4eed-9c2c-b15ea1430d10.png)

Now, to demonstrate that it's not my model, let's take it into Unreal. Here's the model in Unreal:

![tile5](https://user-images.githubusercontent.com/36735507/156333745-0652d3fb-a94c-4d29-a248-aab1eedf0369.png)

Notice - no stretching of the diamond pattern on the edge, just like in the MAX importer before it gets into MAX proper. Now, let's try putting two of them together to see if they tile:

![tile6](https://user-images.githubusercontent.com/36735507/156333893-f4797253-5dbd-401f-aa88-b0f35e61374a.png)

As you can see, they tile perfectly. So, the problem is not my model. Something is going on with MAX once imported, causing things to not line up properly between the material and the UV map. I suspect that the material has shifted a pixel or so? But it's enough to cause issues if someone is trying to create perfectly tiling models like this.

**Comment by ArgentArts** [comment]
If you need to see the model and textures for this, I can zip them up. Just let me know.

**Comment by ArgentArts** [comment]
Don't know if you've worked on this, but the issue seems to have resolved itself with this last update:

![tile_fixed](https://user-images.githubusercontent.com/36735507/156850407-51bf804f-8a56-4761-9fbd-11ec7b4a4411.png)

**Comment by ArgentArts** [comment]
Nope. It does not appear to have been fixed. It just sort of shows up randomly. Here are screen shots of a revised floor tile I'd made in the MAX importer:

![Tile7](https://user-images.githubusercontent.com/36735507/156893100-7a5ae1e5-b5fa-4d6a-b062-e9fd0f88a0af.png)
![Tile8](https://user-images.githubusercontent.com/36735507/156893103-abf69481-6f5c-4b86-b1bf-d50b969d5445.png)
![Tile9](https://user-images.githubusercontent.com/36735507/156893104-11d6ec0f-0ddc-466d-9428-6255779f36ed.png)
 
It looks as it should. All textures are displaying as per their UVs. No issues as I go around the model. But once the model is brought into MAX actual, that all changes:

![Tile10](https://user-images.githubusercontent.com/36735507/156893138-b46c6c97-9c73-4d3e-8b59-cc7cc060b54c.png)
![Tile11](https://user-images.githubusercontent.com/36735507/156893140-c75cde75-4c11-4785-9ee5-aac5e96fe653.png)
![Tile12](https://user-images.githubusercontent.com/36735507/156893146-714d2b15-6528-45b1-8c83-5b411531f2de.png)

This makes it unreliable to use MAX when importing our own models.

**Comment by ArgentArts** [comment]
Totally redid the UVs (provided more space between UV islands, even), redid the textures, and here's the result:

In the MAX importer:
![Tile13](https://user-images.githubusercontent.com/36735507/156895332-5544938c-a1d8-4415-9cf0-4f08e6c244a8.png)

It looks perfect, just like it should.

In MAX proper:
![Tile14](https://user-images.githubusercontent.com/36735507/156895346-9a7d3036-926f-4c2d-951a-3e3410a846fe.png)

It looks totally messed. Ugh. I can't move forward with my level.

**Comment by ArgentArts** [comment]
Here's the same tile in Unreal (for reference):

![Tile15](https://user-images.githubusercontent.com/36735507/156895575-d873d13f-d663-4a31-a68a-0b61e391afa8.png)

**Comment by ArgentArts** [comment]
Okay! Some resolution! I found the cause of the issue. It's MAX, but it's not because it's causing UVs or textures to slip. For whatever reason, there were remnants of materials, .dds, in the entitybank. They weren't named after my model names, but they had one of my old model's texture layouts. For whatever reason, MAX was grabbing this when my model left the exporting and it came into MAX. I found several of these "ghosts" in the entitybank folder. Once I deleted them all and reimported my model, it came in perfectly. 

![Tile16](https://user-images.githubusercontent.com/36735507/156898681-6fd12c16-4006-4b2c-ab02-100527e25547.png)

I am sorry that I did not save the strange "ghost" .dds files to see what they were named. I only recognized them by the pattern of the texture when I saw them going through each file in the entity bank folder. I didn't notice them when I was looking to delete my old models by name. If this occurs again, I will note the file names.

**Comment by ArgentArts** [comment]
I'm going to close this because the issue was my own fault. Sigh. It turns out I wasn't setting up Substance Painter properly and I wasn't paying attention. It was spitting out my textures with the same name (default_normal, default_ao, etc.) and even though I had created a unique material name for the model in Blender, when I imported the FBX into MAX, it dragged in all the correct textures (with the first name "default") and created a .dds file with the first name "default". Then, when I would create a new model or revise a model, these would also get the "default" name for they would draw from the old one in the entitybank directory instead of generating a new one. And THAT was why the UVs were messed up ... because the new test model was slightly different. Now that I've caught my (very stupid) mistake, this won't happen again. I am very sorry to have wasted your time. But the BUG was MINE. Not MAX's.
